### PR TITLE
cliccl: add `load show` subcommand `incremental` to display incremental backups

### PR DESCRIPTION
**Pull request by Elliebababa**
Previously, `load show` only supports showing metadata of a single backup
manifest. Users are ignorant of the incremental backups and have no ways to
inspect the incremental backup manifests unless they know the incremental paths
beforehand (by running `show backup` in a sql session).

This PR updates `load show` with `incremental` subcommand to display
incremental backups, i.e. displaying the auto appended incremental backup
paths of a backup. After listing the incremental paths, users are able to
inspect an incremental backup manifest with `load show summary`.

see #61131

Release note (cli change): Update `load show` with `incremental` subcommand to display 
incremental backups. User can inspect incremental backup chain by running
`cockroach load show incremental <backup_url>.` After listing the incremental paths, 
users are able to inspect an incremental backup manifest with `load show summary`.

```
$ ./cockroach load show incremental "nodelocal://0/test_inc"
./                          -                           2021-03-10T15:48:07Z
./20210310/154947.85        2021-03-10T15:48:07Z        2021-03-10T15:49:47Z
./20210310/155910.43        2021-03-10T15:49:47Z        2021-03-10T15:59:10Z
```

### CODE REVIEW

File: pkg/ccl/backupccl/manifest_handling.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ import (
 const (
 	// backupManifestName is the file name used for serialized BackupManifest
 	// protos.
+	BackupManifestName = "BACKUP_MANIFEST"
 	backupManifestName = "BACKUP_MANIFEST"
```
Comment (pbardea) [inline pkg/ccl/backupccl/manifest_handling.go:49]: I think we can export `BackupManifestName` instead? I actually think we can avoid exporting it if we use `readBackupManifestFromStore` instead of `readBackupManifest`.

File: pkg/ccl/cliccl/load_test.go [hunk 2]
```diff
@@ -102,3 +108,55 @@ Tables:
...
+
+	sqlDB := sqlutils.MakeSQLRunner(db)
+	sqlDB.Exec(t, `CREATE DATABASE testDB`)
+	sqlDB.Exec(t, `USE testDB`)
+	const backupPath = "nodelocal://0/fooFolder"
+	initTS := hlc.Timestamp{WallTime: timeutil.Now().UnixNano()}
+	initAOST := initTS.AsOfSystemTime()
+	sqlDB.Exec(t, fmt.Sprintf(`BACKUP DATABASE testDB TO $1 AS OF SYSTEM TIME '%s'`, initAOST), backupPath)
+
+	// we do two more incremental backups on the full backups.
```
Comment (pbardea) [inline pkg/ccl/cliccl/load_test.go:133]: Nit: Could we capitalize this comment?

File: pkg/ccl/cliccl/load_test.go [hunk 3]
```diff
@@ -102,3 +108,55 @@ Tables:
...
+	expectedIncFolder2 := incTS2.GoTime().Format(backupccl.DateBasedIncFolderName)
+
+	var buf bytes.Buffer
+	w := tabwriter.NewWriter(&buf, 28, 1, 2, ' ', 0)
+	fmt.Fprintf(w, "./	-	%s\n", initTS.GoTime().Format(time.RFC3339))
+	fmt.Fprintf(w, ".%s	%s	%s\n", expectedIncFolder, initTS.GoTime().Format(time.RFC3339), incTS.GoTime().Format(time.RFC3339))
+	fmt.Fprintf(w, ".%s	%s	%s\n", expectedIncFolder2, incTS.GoTime().Format(time.RFC3339), incTS2.GoTime().Format(time.RFC3339))
+	if err := w.Flush(); err != nil {
+		t.Errorf("TestLoadShowIncremental: flush: %v", err)
+		return
```
Comment (pbardea) [inline pkg/ccl/cliccl/load_test.go:153]: Do we need this error after the `t.Errorf`?

File: pkg/ccl/backupccl/manifest_handling.go [hunk 4]
```diff
@@ -71,8 +72,8 @@ const (
 const (
 	// BackupFormatDescriptorTrackingVersion added tracking of complete DBs.
 	BackupFormatDescriptorTrackingVersion uint32 = 1
-
-	dateBasedIncFolderName  = "/20060102/150405.00"
+	// DateBasedIncFolderName exported for cliccl backup inspection
```
Comment (pbardea) [inline pkg/ccl/backupccl/manifest_handling.go:75]: Since we're documenting this here, I think we could also add something like:
```
// DateBasedIncFolderName is the date format used when creating sub-directories
// storing incremental backups for auto-appendable backups.
// It is exported for testing backup inspection tooling.
```

File: pkg/ccl/backupccl/manifest_handling.go [hunk 5]
```diff
@@ -180,6 +181,17 @@ func decompressData(descBytes []byte) ([]byte, error) {
 	return ioutil.ReadAll(r)
 }
 
+// ReadBackupManifest is the export version of readBackupManifest,
+// used in cliccl backup inspection.
+func ReadBackupManifest(
```
Comment (pbardea) [inline pkg/ccl/backupccl/manifest_handling.go:186]: I think that using `readBackupManifestFromStore` (and exporting it) would be what we want here. Old backup manifests may have a different file name, and that method checks handling both.

File: pkg/ccl/cliccl/load.go [hunk 6]
```diff
@@ -122,6 +135,55 @@ func runLoadShowSummary(cmd *cobra.Command, args []string) error {
...
+	if err != nil {
+		return err
+	}
+
+	incPaths, err := backupccl.FindPriorBackupLocations(ctx, store)
+	if err != nil {
+		if errors.Is(err, cloudimpl.ErrListingUnsupported) {
+			// If we do not support listing, we have to just assume there are none
+			// and show the specified base.
+			log.Warningf(ctx, "storage sink %T does not support listing, only resolving the base backup", store)
```
Comment (pbardea) [inline pkg/ccl/cliccl/load.go:156]: Can we actually log like this? Do you know where these logs go? I think we'll probably want to surface this more in the CLI tool (as opposed to when we were taking the backup). Perhaps we could surface this as an error to the user since this command won't work against these stores?

File: pkg/ccl/cliccl/load.go [hunk 7]
```diff
@@ -122,6 +135,55 @@ func runLoadShowSummary(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+func runLoadShowIncremental(cmd *cobra.Command, args []string) error {
+
+	path := args[0]
+	if !strings.Contains(path, "://") {
```
Comment (pbardea) [inline pkg/ccl/cliccl/load.go:141]: Is this converting a path like `/path/to/my/file` to `nodelocal://0/path/to/my/file`? If so, I think we might want to support paths to files still with an explicit scheme so the argument stays a URI. Something like: `file://path/to/my/file` perhaps? This change may be large enough that it warrants it's own PR.

**Review by pbardea — COMMENTED** [review]
(Just looking at the last commit, which is why I'm using GH comments instead of reviewable).

This is looking pretty good!

File: pkg/ccl/cliccl/load.go (hunk 7 above)
Comment (Elliebababa) [reply pkg/ccl/cliccl/load.go:140]: Agree. Will follow up in another PR.

Comment (pbardea) [reply pkg/ccl/cliccl/load.go:140]: Do we need to keep this if statement around until we add support for that? It doesn't look like we have any tests for `./cockroach load show incremental /path/to/backup`?

File: pkg/ccl/cliccl/load_test.go [hunk 8]
```diff
@@ -102,3 +109,54 @@ Tables:
...
+	incAOST2 := incTS2.AsOfSystemTime()
+	sqlDB.Exec(t, fmt.Sprintf(`BACKUP DATABASE testDB TO $1 AS OF SYSTEM TIME '%s'`, incAOST2), backupPath)
+
+	out, err := c.RunWithCapture(fmt.Sprintf("load show incremental %s --external-io-dir=%s", backupPath, dir))
+	require.NoError(t, err)
+	expectedIncFolder := incTS.GoTime().Format(backupccl.DateBasedIncFolderName)
+	expectedIncFolder2 := incTS2.GoTime().Format(backupccl.DateBasedIncFolderName)
+
+	var buf bytes.Buffer
+	w := tabwriter.NewWriter(&buf, 28, 1, 2, ' ', 0)
```
Comment (pbardea) [inline pkg/ccl/cliccl/load_test.go:148]: nit: Can we add some comments here like `/* minWidth */` and `/* tabWidth */` for these arguments?

**Review by pbardea — APPROVED** [review]
LGTM with nits
